main.py

```diff
@@ -222,31 +230,56 @@ def change_visibility(region_id, image_id, public=False):
-    f = open(file_path)
-    images = json.load(f)
-    for image in images:
-        image_id = image['image_id']
-        region_id = image['region_id']
-        if check_tag_key is not None and check_tag_value is not None:
-            logging.debug(f"Checking for {check_tag_key}={check_tag_value} before deleting {image_id}")
-            image_info = get_image_info(region_id, image_id)
-            for tag in image_info['Images']['Image'][0]['Tags']['Tag']:
-                if tag['TagKey'] == check_tag_key and tag['TagValue'] == check_tag_value:
-                    logging.warning(f"{image_id} is tagged with {check_tag_key}={check_tag_value}; will not delete")
-                    # return empty JSON doc
-                    return json.load("{}")
-
-    logging.debug(f"Going to delete {image_id} in {region_id}")
-    client = create_client(region_id)
-    delete_req = DeleteImageRequest()
-    delete_req.set_ImageId(image_id)
-    delete_req.set_protocol_type('https')
+    if check_tag_key is None:
+        check_tag_key = "bootimage"
+    if check_tag_value is None:
+        check_tag_value = "true"
 
-    logging.warning(f"Deleting {image_id} in {region_id}")
-    delete_req = run_cmd([client, delete_req])
-    if delete_req  == 'dry_run':
-        return
-    return json.loads(delete_resp.decode("utf-8"))
+    if os.path.exists(file_path):
+        logging.debug("Found file")
+        with open(file_path, 'r') as f:
+            deleted_images_json = json.load(f)
+    for buildid in deleted_images_json.keys():
+        # enumerate the list of regions/images
+        for pos, item in enumerate(deleted_images_json[buildid]):
+            region_id = item['region']
+            image_id = item['image']
+            # if the image hasn't been marked deleted, remove it, and then update
+            # the 'deleted' key to True
+            if not item["deleted"]:
+                logging.warning(f"Deleting {image_id} in {region_id}")
+                # TODO: uncomment these when we want to go live
+                # we have to mark the image private before deleting it
+
+                if check_tag_key is not None and check_tag_value is not None:
+                    logging.debug(f"Checking for {check_tag_key}={check_tag_value} before deleting {image_id}")
+                    image_info = get_image_info(region_id, image_id)
+                    for tag in image_info['Images']['Image'][0]['Tags']['Tag']:
+                        if tag['TagKey'] == check_tag_key and tag['TagValue'] == check_tag_value:
+                            logging.warning(f"{image_id} is tagged with {check_tag_key}={check_tag_value}; will not delete")
+                            # return empty JSON doc
+                            return json.load("{}")
+                        if image_info['Images']['Image'][0]['IsPublic'] is True:
+                             pass
+                             #change_visibility(region_id image_id, public=False)
```

Comment:
Let's just delete this branch of code; it's a fine idea, but I don't think we need to use it in practice

I think we'll have enough checks before we decide to delete any images

Reply:
Well, we have to keep the call to `change_visibility()`